public/app/core/components/UserEdit/Admin/UserOrganizations.tsx

```diff
@@ -0,0 +1,194 @@
+    this.setState({ userOrgs });
+  }
+
+  render() {
+    const { user } = this.props;
+    const { newOrgRole, roles, userOrgs } = this.state;
+    return (
+      <>
+        <h3 className="page-sub-heading">Organizations</h3>
+        <form name="addOrgForm" className="gf-form-group">
+          <div className="gf-form-inline">
+            <div className="gf-form">
+              <FormLabel>Add</FormLabel>
+              <OrganizationPicker className="max-width-20" onSelected={this.onOrgSelect} />
+            </div>
+            <div className="gf-form">
+              <FormLabel>Role</FormLabel>
+              <Select
+                inputId="role-picker"
+                className="width-10"
+                value={roles.find(role => role.value === newOrgRole)}
+                options={roles}
+                onChange={role => this.onRoleSelect(role.value)}
+              />
+            </div>
+            <div className="gf-form">
+              <Button
+                onClick={event => {
+                  event.preventDefault();
+                  this.onAddToOrg();
+                }}
+              >
+                Add
+              </Button>
+            </div>
+          </div>
+        </form>
+        <div className="gf-form-group">
+          <table className="filter-table form-inline">
+            <thead>
+              <tr>
+                <th>Name</th>
+                <th>Role</th>
+                <th />
+              </tr>
+            </thead>
+            <tbody>
+              {userOrgs.map((userOrg: UserOrg, index) => {
+                return (
+                  <tr key={index}>
+                    <td>
+                      {userOrg.name} {userOrg.orgId === user.orgId && <span className="label label-info">Current</span>}
+                    </td>
+                    <td>
+                      <Select
+                        inputId={userOrg.orgId + '-role-picker'}
+                        className="width-10"
+                        value={roles.find(role => role.value === userOrg.role)}
+                        options={roles}
+                        onChange={role => this.onOrgRoleChange(userOrg, role.value)}
+                      />
+                    </td>
+                    <td>
+                      <Button
+                        variant="danger"
+                        size="sm"
+                        onClick={() => {
+                          this.removeFromOrg(userOrg);
+                        }}
+                      >
+                        <i className="fa fa-power-off" />
+                      </Button>
+                    </td>
+                  </tr>
+                );
+              })}
+            </tbody>
+          </table>
+        </div>
+      </>
+    );
+  }
+}
+
+function mapStateToProps(state: StoreState) {
+  return {
+    user: state.user.profile,
+  };
+}
+
+const mapDispatchToProps = {};
+
+export default hot(module)(
```

Comment:
since this is a component used by a page component try to use the redux only on the top level component (if feasible) This component can just take in the OrgUser as a prop instead. Makes the component a lot easier to reason about, having "hidden" redux hooks deep in the tree makes it a lot harder on the parent level to understand component data flows and where they get their data from. If this was a prop instead if would be super clear (and then we can remove userId as prop as well).